core/src/bundle/usBundleContext.cpp

```diff
@@ -169,9 +176,12 @@ struct ServiceHolder
     {
       bc->UngetService(sref);
     }
-    catch (const std::exception& exp)
+    catch (const std::exception& )
     {
-      US_INFO << "UngetService threw an exception - " << exp.what();
+	  // don't throw exceptions from the destructor. For an explanation, see:
+	  // https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md
+	  // Following this rule means that a FrameworkEvent isn't an option here 
+	  // since it contains an exception object which clients could throw.
```

Comment:
This wasn't throwing before, was it? What about keeping a log message here, similar to the previous US_INFO call?